### ISSUE TITLE
Error message(s) for required fields

### ISSUE THREAD

**Issue by valianttry** [mozilla/donate.mozilla.org#673]
Outcome:
I hit "submit donation" without filling out any fields and they required fields all turn red, but the error only mentions the privacy policy.

Expected outcome:
I hit "submit donation" and an error message says something like "please complete required fields in red"

![donate_mozilla_org___give_to_mozilla_today](https://cloud.githubusercontent.com/assets/9830371/10952374/3aebcfbc-82f7-11e5-9ae7-10273949d957.png)

**Comment by adamlofting** [comment]
Related: #412 #413 #485

**Comment by ScottDowne** [comment]
I have been in a case before where designers have had me add an error message to every field...

This is the project: https://petitions.mozilla.org/data-retention/

I previously had one error under email and it got nixed during QA.

The argument was adding a "fields highlighted in red are required"-ish error under email was confusing and too strongly associated with email.

I was kinda hoping a red highlight for required errors was clear enough and an error message wasn't needed.

**Comment by ScottDowne** [comment]
I did a bit of research on this, for fun.

It's clear error messages are not clear... Lots of opinions.

Some bits I learned that I like.

1. Put errors close to the point of input. I read this two ways, but more on that in a bit.
2. Put errors above point of input. This is because screen readers and people read top down.

Sooo, I think a fairly good solution for now is put the error above the submit button, which because we do validation on submit, and not as the user types, it means our point of input is the button itself, the button has the user's attention when they click.
